### PR TITLE
Support "local ps" to list running local ECS task containers

### PR DESCRIPTION
**Pull request by efekarakus**
**Description of changes**: Lists the running containers part of the ECS tasks running locally.

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**  
- [x] Unit tests passed
- [x] Integration tests passed
- [ ] Unit tests added for new functionality
- [x] Listed manual checks and their outputs in the comments ([example](https://github.com/aws/amazon-ecs-cli/pull/750#issuecomment-472623042))
- [x] Integration tests added for new functionality

**Documentation**  
- [N/A] Contacted our doc writer
- [N/A] Updated our README
----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by efekarakus** [comment]
## Manual Tests

#### 1. Nothing running locally
```shell
$ dev-ecs-cli local ps         
CONTAINER ID        IMAGE               STATUS              PORTS               NAMES               TASKDEFINITIONARN   TASKFILEPATH
$ dev-ecs-cli local ps --json           
[]
```

#### 2. With a task container running
```shell
$ dev-ecs-cli local ps     
CONTAINER ID        IMAGE                 STATUS              PORTS                  NAMES                    TASKDEFINITIONARN                                               TASKFILEPATH
2ddaeed688aa        ecs-local-other_app   Up 2 seconds        0.0.0.0:6000->80/tcp   /ecs-local-other_app_1   arn:aws:ecs:us-east-1:685593908319:task-definition/efe-test:1  
$ dev-ecs-cli local ps --json                                                                                                                                                                                                       local/ps ⬆ ✭
[
  {
    "Id": "2ddaeed688aac5297d13dd1dc2dda4e9a8a1baebeb4b4fc2198da783e122a023",
    "Names": [
      "/ecs-local-other_app_1"
    ],
    "Image": "ecs-local-other_app",
    "ImageID": "sha256:0ac30848f74a68b442cee205df0b20e8a038bb061776a1786363510dbc2c6f85",
    "Command": "/bin/sh -c '/bin/bash -c 'while true; do sleep 30; done;''",
    "Created": 1558723259,
    "Ports": [
      {
        "IP": "0.0.0.0",
        "PrivatePort": 80,
        "PublicPort": 6000,
        "Type": "tcp"
      }
    ],
    "Labels": {
      "ECSLocalTask": "True",
      "com.docker.compose.config-hash": "1d67f24db5737d34353d6112e9cf51d9fd5713e7e52eae81eb6c0bd9873caafc",
      "com.docker.compose.container-number": "1",
      "com.docker.compose.oneoff": "False",
      "com.docker.compose.project": "ecs-local-other",
      "com.docker.compose.service": "app",
      "com.docker.compose.version": "1.23.2",
      "taskDefinitionARN": "arn:aws:ecs:us-east-1:685593908319:task-definition/efe-test:1"
    },
    "State": "running",
    "Status": "Up 34 seconds",
    "HostConfig": {
      "NetworkMode": "ecs-local-network"
    },
    "NetworkSettings": {
      "Networks": {
        "ecs-local-network": {
          "IPAMConfig": null,
          "Links": null,
          "Aliases": null,
          "NetworkID": "cb43b4445f11ac4e8f337f68904e3d1eb2b900b80deabe4c62cea56c063ad029",
          "EndpointID": "a4b718d0cc53f171e5e5220f81d6647d24990724b7439d78be0438c2decdab97",
          "Gateway": "169.254.170.1",
          "IPAddress": "169.254.170.3",
          "IPPrefixLen": 24,
          "IPv6Gateway": "",
          "GlobalIPv6Address": "",
          "GlobalIPv6PrefixLen": 0,
          "MacAddress": "02:42:a9:fe:aa:03",
          "DriverOpts": null
        }
      }
    },
    "Mounts": []
  }
]
```

File: ecs-cli/modules/cli/local/ps_app.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/client"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+	"golang.org/x/net/context"
+)
+
+// TODO These labels should be defined part of the local.Create workflow.
+// Refactor to import these constants instead of re-defining them here.
+const (
```
Comment (efekarakus) [inline ecs-cli/modules/cli/local/ps_app.go]: @SoManyHs These are the Docker object labels that I'm expecting to be part of each container. Let me know if you're planing to use a different value for them :)

File: ecs-cli/integ/e2e/local_test.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+	type commandTest struct {
+		args    []string
+		execute func(t *testing.T, args []string)
+	}
+
+	tests := map[string]struct {
+		sequence []commandTest
+	}{
+		"clean state": {
```
Comment (PettitWesley) [inline ecs-cli/integ/e2e/local_test.go]: Is there a reason why this is a table driven test? 

Looks like it only has one test case...

File: ecs-cli/modules/cli/local/create_app.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+	"fmt"
+
+	"github.com/urfave/cli"
+)
+
+func Create(c *cli.Context) {
+	// 1. read in task def (from file or arn)
+	// 2. parse task def into go object
+	// 3. write to docker-compose.local.yml file
+	fmt.Println("foo") // placeholder
```
Comment (PettitWesley) [inline ecs-cli/modules/cli/local/create_app.go]: lol

File: ecs-cli/integ/e2e/local_test.go (hunk 2 above)
Comment (efekarakus) [reply ecs-cli/integ/e2e/local_test.go]: soon™ we'll have full ecs-cli local workflows as integration tests :P.

This sequence for example will make up on test case: "ecs-cli local create --arn" -> "ecs-cli local up" -> "ecs-cli local ps" -> "ecs-cli local stop"

File: ecs-cli/modules/cli/local/docker.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import (
+	"os"
+
+	"github.com/docker/docker/client"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// minDockerAPIVersion is the oldest Docker API version supporting the operations used by "local" sub-commands.
+	minDockerAPIVersion = "1.27"
```
Comment (PettitWesley) [inline ecs-cli/modules/cli/local/docker.go]: How'd you pick this version?

Comment (efekarakus) [reply ecs-cli/modules/cli/local/docker.go]: I picked the same version that the endpoints container requires. The operations that we use so far in "ecs local" are all supported in [v1.27](https://docs.docker.com/engine/api/v1.27/).

I can say that the endpoints container uses 1.27 and link to the code ?

File: ecs-cli/integ/e2e/local_test.go (hunk 2 above)
Comment (PettitWesley) [reply ecs-cli/integ/e2e/local_test.go]: 😎

File: ecs-cli/modules/cli/local/docker.go (hunk 4 above)
Comment (PettitWesley) [reply ecs-cli/modules/cli/local/docker.go]: Sure 👍🏻

Comment (efekarakus) [reply ecs-cli/modules/cli/local/docker.go]: Done!

Comment (PettitWesley) [reply ecs-cli/modules/cli/local/docker.go]: Actually I'd prefer it if the comment also said that all the operations in ecs local are supported in v1.27

Its hypothetically possible that the "ecs local" commands would need a newer Docker client version than the endpoints container.

The current comment might lead future maintainers to think that the commands and the endpoints container have to use the same Docker Client version. And that's not true.

Comment (efekarakus) [reply ecs-cli/modules/cli/local/docker.go]: 👍 makes sense, done!

**Pull request merged**